---
title: "DG25-31: Some users might be blocked from accessing defguard via OpenID"
severity: Info
cvss_string: "/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:N"
issue_link: https://github.com/DefGuard/defguard/issues/1549
---

### Technical details

1.  Username test.test had been created. Email phtest2@isec.pl had been
    assigned to him.

2.  Different user - test.test@isec.pl wants to log in via OpenID.

3.  OpenID extracts test.test from test.test@isec.pl and tries to create
    such username.

4.  Since test.test username is already registered (step 1) - API throws
    an error and legitimate user test.test@isec.pl cannot log in.

**Request:**\
\
**POST** /api/v1/openid/callback HTTP/2\
**Host:** defguard.dvpnsec.net\
\[\...\]\
\
{\"code\":\"\<cut\>\",\"state\":\"\<cut\>\"}\
\
\
**Response:**\
\
**HTTP/2** 401 Unauthorized\
**Alt-Svc:** h3=\":443\"; ma=2592000\
**Content-Type:** application/json\
**Date:** Fri, 29 Aug 2025 11:41:15 GMT\
**Server:** Caddy\
**X-Defguard-Version:** 1.5.0-a29ac10\
**Content-Length:** 61\
\
{\"msg\":\"User with username test.test already exists\"}
